spectator/registry.go

```diff
@@ -0,0 +1,214 @@
+	r.config.Log = logger
+}
+
+// TODO investigate how to provide the same functionality for MeterFactory. Do we need to expose the writer?
+// NewMeter creates a new Meter using the provided meterFactory
+func (r *Registry) NewMeter(id *meter.Id, meterFactory MeterFactoryFun) Meter {
+	return meterFactory()
+}
+
+// NewId calls spectator.NewId().
+func (r *Registry) NewId(name string, tags map[string]string) *meter.Id {
+	return meter.NewId(name, tags)
+}
+
+// TODO append common tags upon creation
+
+// Counter calls NewId() with the name and tags, and then calls r.CounterWithId()
+// using that *Id.
+func (r *Registry) Counter(name string, tags map[string]string) *meter.Counter {
+	return meter.NewCounter(meter.NewId(name, tags), r.writer)
+}
+
+func (r *Registry) CounterWithId(id *meter.Id) *meter.Counter {
+	return meter.NewCounter(id, r.writer)
+}
+
+// Timer calls NewId() with the name and tags, and then calls r.TimerWithId() using that *Id.
+func (r *Registry) Timer(name string, tags map[string]string) *meter.Timer {
+	return meter.NewTimer(meter.NewId(name, tags), r.writer)
+}
+
+// TimerWithId returns a new *Timer, using the provided meter identifier.
+func (r *Registry) TimerWithId(id *meter.Id) *meter.Timer {
+	return meter.NewTimer(id, r.writer)
+}
+
+func (r *Registry) Gauge(name string, tags map[string]string) *meter.Gauge {
+	return meter.NewGauge(meter.NewId(name, tags), r.writer)
+}
+
+func (r *Registry) GaugeWithId(id *meter.Id) *meter.Gauge {
+	return meter.NewGauge(id, r.writer)
+}
+
+func (r *Registry) MaxGauge(name string, tags map[string]string) *meter.MaxGauge {
+	return meter.NewMaxGauge(meter.NewId(name, tags), r.writer)
+}
+
+func (r *Registry) MaxGaugeWithId(id *meter.Id) *meter.MaxGauge {
+	return meter.NewMaxGauge(id, r.writer)
+}
+
+func (r *Registry) DistributionSummary(name string, tags map[string]string) *meter.DistributionSummary {
+	return meter.NewDistributionSummary(meter.NewId(name, tags), r.writer)
+}
+
+func (r *Registry) DistributionSummaryWithId(id *meter.Id) *meter.DistributionSummary {
+	return meter.NewDistributionSummary(id, r.writer)
+}
+
+func (r *Registry) PercentileDistributionSummary(name string, tags map[string]string) *meter.PercentileDistributionSummary {
+	return meter.NewPercentileDistributionSummary(meter.NewId(name, tags), r.writer)
+}
+
+func (r *Registry) PercentileDistributionSummaryWithId(id *meter.Id) *meter.PercentileDistributionSummary {
+	return meter.NewPercentileDistributionSummary(id, r.writer)
+}
+
+func (r *Registry) PercentileTimer(name string, tags map[string]string) *meter.PercentileTimer {
+	return meter.NewPercentileTimer(meter.NewId(name, tags), r.writer)
+}
+
+func (r *Registry) PercentileTimerWithId(id *meter.Id) *meter.PercentileTimer {
+	return meter.NewPercentileTimer(id, r.writer)
+}
+
+// Stop shuts down the running goroutine(s), and attempts to flush the metrics.
+func (r *Registry) Stop() {
+	// TODO implement
+	close(r.quit)
+}
+
+// TODO do we need this?
```

Comment:
Not sure if we need to filter out NaN measurements. 
I don't think other implementations are doing it. 
My initial thinking is that it's not require because we're not operating on the values passed in, but I need to investigate further

Reply:
Confirming that we don't need this code because we already check for positive values on each of the Meters and since we are not operating on the values we can't get NaN.
Will remove